lib/build.py

```diff
@@ -250,10 +250,16 @@ def __init__(self, tree):
       for st in ch.tree_children(tree, "option"):
          k = ch.tree_terminal(st, "OPTION_KEY")
          v = ch.tree_terminal(st, "OPTION_VALUE")
-         if (k in self.options):
-            ch.FATAL("%3d %s: repeated option --%s"
-                     % (self.lineno, self.str_name, k))
-         self.options[k] = v
+         if (k == "arg"):
+            try:
+               self.options[k] = self.options[k] + "=" + v
+            except KeyError:
+               self.options[k] = v
+         else:
+            if (k in self.options):
+               ch.FATAL("%3d %s: repeated option --%s"
+                        % (self.lineno, self.str_name, k))
+            self.options[k] = v
```

Comment:
I think this logic might be better in the grammar. It could grow two flavors of options: normal (what we had before) and key-value (or a better name?), that have a value that itself consists of a key/value pair. Then all this code has to do is pick out things from the parse.